kv/kvnemesis: TestKVNemesisMultiNode_Partition_Safety failed

kv/kvnemesis.TestKVNemesisMultiNode_Partition_Safety [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20804811?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20804811?buildTab=artifacts#/) on master @ [4bcccc8118deed40d6056478edc036542726c297](https://github.com/cockroachdb/cockroach/commits/4bcccc8118deed40d6056478edc036542726c297):


```
=== RUN   TestKVNemesisMultiNode_Partition_Safety
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/logTestKVNemesisMultiNode_Partition_Safety1651888388
    test_log_scope.go:82: use -show-logs to present logs inline
    kvnemesis_test.go:631: seed: 3066197698605002853
    kvnemesis_test.go:277: n1 using clock[offset=207.438µs,amp=3.292562ms]
    kvnemesis_test.go:277: n2 using clock[offset=225.831µs,amp=3.274169ms]
    kvnemesis_test.go:277: n3 using clock[offset=-334.998µs,amp=3.165002ms]
    kvnemesis_test.go:277: n4 using clock[offset=129.206µs,amp=3.370794ms]
    kvnemesis_test.go:354: kvnemesis logging to /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/kvnemesis4081599895
    kvnemesis.go:240: error applying env.UpdateZoneConfig(ctx, ToggleGlobalReads) // pq: replica unavailable: (n1,s1):1 unable to serve request to r34:/NamespaceTable/{30-Max} [(n1,s1):1, (n4,s4):2, (n2,s2):3, next=4, gen=4]: closed timestamp: 1764774052.693822756,0 (2025-12-03 15:00:52); raft status: {"id":"1","term":6,"vote":"1","commit":29,"lead":"0","leadEpoch":"0","raftState":"StatePreCandidate","applied":29,"progress":{},"leadtransferee":"0"}: replica has been leaderless for 1m0s: pq: replica unavailable: (n1,s1):1 unable to serve request to r34:/NamespaceTable/{30-Max} [(n1,s1):1, (n4,s4):2, (n2,s2):3, next=4, gen=4]: closed timestamp: 1764774052.693822756,0 (2025-12-03 15:00:52); raft status: {"id":"1","term":6,"vote":"1","commit":29,"lead":"0","leadEpoch":"0","raftState":"StatePreCandidate","applied":29,"progress":{},"leadtransferee":"0"}: replica has been leaderless for 1m0s
    kvnemesis.go:281: failures(verbose): /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/kvnemesis4081599895/failures
        repro steps: /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/kvnemesis4081599895/repro.go
        rangefeed KVs: /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/kvnemesis4081599895/kvs-rangefeed.txt
        scan KVs: /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/kvnemesis4081599895/kvs-scan.txt
    kvnemesis_test.go:755: Metric                              | Node 1                         | Node 2                         | Node 3                         | Node 4                        
    kvnemesis_test.go:761: ------------------------------------+--------------------------------+--------------------------------+--------------------------------+-------------------------------
    kvnemesis_test.go:768: follower_reads.success_count        | 0                              | 2                              | 9                              | 3                             
    kvnemesis_test.go:768: raft.commands.proposed              | 4143                           | 710                            | 661                            | 335                           
    kvnemesis_test.go:768: raft.commands.reproposed.new-lai    | 0                              | 0                              | 6                              | 1                             
    kvnemesis_test.go:768: raft.commands.reproposed.unchanged  | 694                            | 151                            | 7                              | 205                           
    kvnemesis_test.go:768: txn.aborts                          | 161                            | 18                             | 49                             | 36                            
    kvnemesis_test.go:768: txn.durations                       | μ=1.092s p50=3ms p99=10s       | μ=406ms p50=4ms p99=7.912s     | μ=515ms p50=5ms p99=10s        | μ=1.105s p50=5ms p99=10s      
    kvnemesis_test.go:768: txn.restarts.writetooold            | 0                              | 2                              | 5                              | 0                             
    kvnemesis_test.go:768: txn.restarts.serializable           | 0                              | 0                              | 0                              | 0                             
    kvnemesis_test.go:768: txn.restarts.readwithinuncertainty  | 0                              | 0                              | 2                              | 0                             
    kvnemesis_test.go:768: txn.server_side.1PC.success         | 103                            | 62                             | 0                              | 0                             
    kvnemesis_test.go:768: txnrecovery.failures                | 0                              | 0                              | 0                              | 0                             
    kvnemesis_test.go:768: txnrecovery.successes.aborted       | 0                              | 0                              | 0                              | 0                             
    kvnemesis_test.go:768: txnrecovery.successes.committed     | 0                              | 0                              | 0                              | 0                             
    kvnemesis_test.go:768: txnwaitqueue.deadlocks_total        | 0                              | 0                              | 12                             | 0                             
    kvnemesis_test.go:689: 
        	Error Trace:	pkg/kv/kvnemesis/kvnemesis_test.go:689
        	            				pkg/kv/kvnemesis/kvnemesis_test.go:488
        	Error:      	Should be zero, but was 1
        	Test:       	TestKVNemesisMultiNode_Partition_Safety
        	Messages:   	kvnemesis detected failures
    dump_raft_logs.go:32: dumping raft logs to /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/kvnemesis4081599895/raftlogs
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4aeb9c412c16a003ed8ddc43788d2bba/logTestKVNemesisMultiNode_Partition_Safety1651888388
--- FAIL: TestKVNemesisMultiNode_Partition_Safety (97.90s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode_Partition_Safety.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57442

--- a/pkg/kv/kvnemesis/applier.go
+++ b/pkg/kv/kvnemesis/applier.go
@@ -114,7 +114,8 @@ func exceptConditionFailed(err error) bool {
 }
 
 func exceptReplicaUnavailable(err error) bool {
-	return errors.HasType(err, (*kvpb.ReplicaUnavailableError)(nil))
+	return errors.HasType(err, (*kvpb.ReplicaUnavailableError)(nil)) ||
+		strings.Contains(err.Error(), "replica unavailable")
 }
 
 func exceptContextCanceled(err error) bool {

